### PR TITLE
219 simulation progress

### LINKED ISSUE #219: Display simulation progress and status
Needs https://github.com/yaptide/yaptide/issues/99

### PR DESCRIPTION
**Pull request by ostatni5**
close #219 
obraz

obraz
obraz

### CODE REVIEW

File: src/ThreeEditor/ThreeEditor.tsx [hunk 1]
```diff
@@ -31,8 +31,12 @@ function ThreeEditor(props: ThreeEditorProps) {
 	}, [editor, props.onEditorInitialized]);
 
 	useEffect(() => {
-		props.focus === true && containerEl.current?.focus();
+		if (props.focus === true) {
```
Comment (Derstilon) [inline src/ThreeEditor/ThreeEditor.tsx:34]:
```suggestion
		if (props.focus) {
```

File: src/ThreeEditor/ThreeEditor.tsx [hunk 2]
```diff
@@ -31,8 +31,12 @@ function ThreeEditor(props: ThreeEditorProps) {
...
+			containerEl.current?.focus();
+			editor?.signals.sceneGraphChanged.dispatch();
+		}
 		props.focus === false && containerEl.current?.blur();
```
Comment (Derstilon) [inline src/ThreeEditor/ThreeEditor.tsx:38]: what if the focus is undefined? why focus can be undefined?

File: src/services/AuthService.tsx [hunk 3]
```diff
@@ -121,35 +153,24 @@ const Auth = (props: AuthProps) => {
...
 			.json()
-			.then((response: unknown) => {
+			.then(() => {
 				setUser(null);
 			})
-			.catch((error: HTTPError) => {
-				console.error(error);
-				console.log(error?.response);
-			});
+			.catch((_: HTTPError) => {});
```
Comment (Derstilon) [inline src/services/AuthService.tsx:173]: why did you remove error logging?

**Review by Derstilon — COMMENTED** [review]
Check out my questions

File: src/services/AuthService.tsx (hunk 3 above)
Comment (ostatni5) [reply src/services/AuthService.tsx:173]: Error logging is handled in the callback after every response 
src\services\AuthService.tsx
obraz

File: src/ThreeEditor/ThreeEditor.tsx (hunk 2 above)
Comment (ostatni5) [reply src/ThreeEditor/ThreeEditor.tsx:38]: Because it is an optional property of the component to indicate if ThreeEditor is focused, but is not always necessary.

Comment (ostatni5) [reply src/ThreeEditor/ThreeEditor.tsx]: Now is value required

**Review by Derstilon — APPROVED** [review]
image
I approve this poggers

**Review by grzanka — APPROVED** [review]
LGTM
